### PR TITLE
Rewrite messages and issue description for TooManyFunctions rule - #552

### PR DESCRIPTION
**Pull request by arturbosch**
Please also look at the other linked commit inside #552. I accidentally pushed the changes to master.
This PR adds messages for different TooManyFunctions findings and repairs the build.

Edit: I don't want to rewrite master history as someone may depend on the newest commit. Reverting the last commit can still be done if the changes are not good enought.

### CODE REVIEW

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/TooManyFunctions.kt [hunk 1]
```diff
@@ -20,9 +20,9 @@ class TooManyFunctions(config: Config = Config.empty) : Rule(config) {
 
 	override val issue = Issue("TooManyFunctions",
 			Severity.Maintainability,
-			"Classes with many functions tend to do too many things and often come in conjunction with " +
-					"large classes and can quickly become God classes. " +
-					"Consider extracting methods to (new) classes better matching their responsibility.")
+			"Too many functions inside a/an file/class/object/interface always indicate a violation of "
+					+ "the single responsibility principle. Maybe your file/class/object/interface wants to manage to many " +
```
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/TooManyFunctions.kt]: Maybe _the_ file/class/...

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/TooManyFunctions.kt [hunk 2]
```diff
@@ -20,9 +20,9 @@ class TooManyFunctions(config: Config = Config.empty) : Rule(config) {
...
+					"things at once. Try to refactor out functionality which belong clearly together.")
```
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/TooManyFunctions.kt]: I'd change the last sentence to "Extract functionality which clearly belongs together."

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/TooManyFunctions.kt [hunk 3]
```diff
@@ -56,20 +57,25 @@ class TooManyFunctions(config: Config = Config.empty) : Rule(config) {
 				report(ThresholdedCodeSmell(issue,
 						Entity.from(klass),
 						Metric("SIZE", amount, thresholdInInterfaces),
-						message = "")) // TODO: add class message
+						"Interface '${klass.name}' with '$amount' functions detected. " +
```
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/TooManyFunctions.kt]: Totally love these messages and the descriptive issue description! 👍

**Comment by arturbosch** [comment]
I included your suggestions, thanks for the reviews!

**Pull request merged**